### PR TITLE
Add templates for various component types

### PR DESCRIPTION
**Pull request by xen0l**
These files serve as a template for creating new components. One should do something like:

<pre>
cd components
mkdir <category>/<component>
cd <category>/<component>
cp ../../../templates/<template> Makefile
</pre>

### CODE REVIEW

File: templates/ruby.mk [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+COMPONENT_SRC=          $(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=      $(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_URL=
+COMPONENT_ARCHIVE_HASH=
+COMPONENT_LICENSE=
+COMPONENT_LICENSE_FILE= $(COMPONENT_NAME).license
+
+include $(WS_MAKE_RULES)/prep.mk
+include $(WS_MAKE_RULES)/setup.py.mk
+include $(WS_MAKE_RULES)/gem.mk
```
Comment (alarcher) [inline templates/ruby.mk:30]: I think you edited the wrong line setup.py -> gem, gem -> ips

**Comment by alarcher** [comment]
This looks very neat. Could you maybe add cmake.mk as well ?

**Comment by xen0l** [comment]
Added, anything else?

**Comment by alarcher** [comment]
Looks good to me.

**Pull request merged**